<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>
	<arg name="model_file" 			default="/srv/data/gait1992/residual_reduction_algorithm/model_adjusted.osim" />
	<arg name="get_second_label" 		default="true"/>
	<arg name="left_foot_tf_name" 		default="left_foot_forceplate" />
	<arg name="right_foot_tf_name" 		default="right_foot_forceplate" />
	<arg name="use_exact_sync"		default="false"/>
	<arg name="max_buffer_length"		default="100"/>

	<!-- If no_rotation is true then I will neglect the reference rotation transformation -->
	<arg name="grf_no_rotation"		default="true"/>

	<node pkg="osrt_ros" type="id_async" 	name="id_node" output="screen">
		<param name="model_file" 		value="$(arg model_file)" />
		<param name="get_second_label" 		value="$(arg get_second_label)" />
		<param name="left_foot_tf_name"		value="$(arg left_foot_tf_name)" />
		<param name="right_foot_tf_name"	value="$(arg right_foot_tf_name)" />
		<!-- each grfm may have its own frame of reference! -->
		<!-- ground :== subject_opensim. maybe we need to change the names to make it more sensible. -->
		<param name="left_reference_frame"        value="ik/subject_opensim"/>
		<param name="right_reference_frame"        value="ik/subject_opensim"/>
		<param name="use_exact_sync"		value="$(arg use_exact_sync)"/>
		<remap from="/id_node/input_filtered" 		to="/ik/output_filtered"/>
		<remap from="in_labels" 			to="/ik/outlabels"/>
		<remap from="out_labels" 			to="/id_node/out_labels"/>
		<!--remap from="/id_node/input2" 	to="/inverse_kinematics_from_file/r_data2"/-->
		<!--remap from="/in_labels2" 	to="/inverse_kinematics_from_file/out_labels"/ -->
		<remap from="/id_node/input2" 	to="/grf_node/output"/>
		<remap from="/in_labels2" 	to="/grf_node/out_labels"/>

		<remap from="/id_node/left/wrench" 	to="/left/wrench_filtered"/>
		<remap from="/id_node/right/wrench" 	to="/right/wrench_filtered"/>
		<param name="grf_right_point_identifier" 		value="ground_force_p" />
		<param name="grf_right_force_identifier" 		value="ground_force_v" />
		<param name="grf_right_torque_identifier" 		value="ground_torque_" />
		<param name="grf_right_apply_to_body" 			value="calcn_r" />
		<param name="grf_right_force_expressed_in_body" 	value="ground" />
		<param name="grf_right_point_expressed_in_body" 	value="ground" />

		<param name="grf_left_point_identifier" 		value="1_ground_force_p" />
		<param name="grf_left_force_identifier" 		value="1_ground_force_v" />
		<param name="grf_left_torque_identifier" 		value="1_ground_torque_" />
		<param name="grf_left_apply_to_body" 			value="calcn_l" />
		<param name="grf_left_force_expressed_in_body" 		value="ground" />
		<param name="grf_left_point_expressed_in_body" 		value="ground" />

		<!-- Filter values for filtering ik (and grfm, the values are currently shared) in cases where we don't use the filtered input -->
		<param name="memory" 			value="35" />
		<param name="cutoff_freq" 		value="6" />
		<param name="delay" 			value="14" />
		<param name="spline_order" 		value="3" />

		<param name="ik_delay"			value="0.26" />
		<param name="max_buffer_length"		value="$(arg max_buffer_length)" />

		<!-- TODO: if I really want to generalize the number of wrenches i also need to change how async gets called-->
		<param name="left_no_rotation"			value="$(arg grf_no_rotation)"/>

		<param name="right_no_rotation"			value="$(arg grf_no_rotation)"/>
	</node>

</launch>
